Sources/protoc-gen-swift/MessageFieldGenerator.swift

```diff
@@ -478,23 +483,17 @@ struct MessageFieldGenerator {
             return
         }
         let storagePrefix = usesHeapStorage ? "_storage." : "self."
-        p.print("\(generatorOptions.visibilitySourceSnippet)var \(swiftHasName): Bool {\n")
-        p.indent()
-        p.print("return \(storagePrefix)\(swiftStorageName) != nil\n")
-        p.outdent()
-        p.print("}\n")
+        p.print("/// Check if `\(swiftName)` has been explicitly set.\n")
+        p.print("\(generatorOptions.visibilitySourceSnippet)var \(swiftHasName): Bool {return \(storagePrefix)\(swiftStorageName) != nil}\n")
     }
 
     func generateClearMethod(printer p: inout CodePrinter, usesHeapStorage: Bool) {
         if isRepeated || isMap || oneof != nil || (isProto3 && !isMessage) {
             return
         }
         let storagePrefix = usesHeapStorage ? "_storage." : "self."
-        p.print("\(generatorOptions.visibilitySourceSnippet)mutating func \(swiftClearName)() {\n")
-        p.indent()
-        p.print("\(storagePrefix)\(swiftStorageName) = nil\n")
-        p.outdent()
-        p.print("}\n")
+        p.print("/// Clear the value of `\(swiftName)`, afterwards it will return its default value.\n")
```

Comment:
``/// Clears the value of `\(swiftName)`. Subsequent reads from it will return its default value.``